www/pages/plan-trip/plan-trip-controller.js

```diff
@@ -217,66 +219,78 @@ angular.module('pvta.controllers').controller('PlanTripController', function ($s
     if (!$scope.params.origin.id || !$scope.params.destination.id) {
       return;
     }
-    $scope.route = {
-      directions: [],
-      arrivalTime: null,
-      departureTime: null,
-      origin: null,
-      destination: null
-    };
+
     // Google won't return trips for times past.
     // Instead of throwing an error, assume the user wants
     // directions for right now.
     if ($scope.params.time.datetime < Date.now()) {
       $scope.updateASAP(true);
     }
+
     $ionicLoading.show({
       template: 'Routing..'
     });
-    // The Trip factory takes care of the forming/sending
-    // the request and retrieving the response for us.
-    Trips.route($scope.params, $scope.directionsDisplay, function (tripDetails) {
-      // Now, we have a crude trip object!
-      $ionicLoading.hide();
-      $scope.route = tripDetails;
-      if ($scope.route.status === google.maps.DirectionsStatus.OK) {
-        // Trip factory will parse the trip and return a clean
-        // array of directions.
-        Trips.generateDirections(function (tripDetails) {
-          $scope.route.directions = tripDetails;
-          $scope.$apply();
-          $scope.scrollTo('route');
-          // Force a map redraw because it was hidden before.
-          // There's an angular bug with ng-show that will cause
-          // the map to draw only grey after being hidden
-          // unless we force a redraw.
-          google.maps.event.trigger($scope.map, 'resize');
-          ga('send', 'event', 'TripStepsRetrieved', 'PlanTripController.getRoute()', 'Received steps for a planned trip!');
-        });
+
+    transitOptions = {
+      modes: [google.maps.TransitMode.BUS]
+    };
+
+  if ($scope.params.time.asap !== true) {
+      if ($scope.params.time.type === 'departure') {
+        transitOptions['departureTime'] = $scope.params.time.datetime;
+      }
+      else if ($scope.params.time.type === 'arrival') {
+        transitOptions['arrivalTime'] = $scope.params.time.datetime;
       }
-      // If Google doesn't have directions for us,
-      // handle it gracefully.
       else {
+        console.error("Determining route for Plan Trip failed due to unexpected input. Expected 'arrival' or 'departure', received" + params.time.type);
```

Comment:
Can you add a Google Analytics event in here?  See [line 270](https://github.com/umts/pvta-multiplatform/pull/245/files#diff-c2cca39ec92cc35caca7992082d316c2R270) for an example.